### PR TITLE
Fix: Floating point imprecision could break vertical text gradient generation

### PR DESCRIPTION
**Pull request by TheWyo**
##### Description of change
In Chrome, I'd noticed some of my text in a project didn't always have the same gradient on each line. Turns out the seams between tiled gradients could sometimes have floating point imprecision, effectively swapping the start/end gradient colours around and messing a line or two up. This PR caps the gradient to the last value (as color stops are generated in order anyway) to prevent this happening.

[Playground demo (Open in chrome, change bool at start to toggle fix)](https://pixiplayground.com/#/edit/8I-EO0eRlShU4YHvuD6bc)

Happy to backport this for v4 as well if wanted.

##### Pre-Merge Checklist

- [ ] Tests and/or benchmarks are included
- Demo provided.
- Do we have a way of testing visual stuff?
- [x] Lint process passed (`npm run lint`)
- [ ] Tests passed (`npm run test`)
- Can't verify, tests currently failing already.

### CODE REVIEW

**Comment by themoonrat** [comment]
Hello @TheWyo ! Thanks for the PR and for the playground
When I try to use your playground, in the console I see

"Uncaught TypeError: Failed to execute 'createLinearGradient' on 'CanvasRenderingContext2D': The provided double value is non-finite."

Could you fix this please? Code looks good, would just like to see for myself :)

File: packages/text/src/Text.ts [hunk 1]
```diff
@@ -503,8 +510,13 @@ export class Text extends Sprite
                     {
                         stop = currentIteration / totalIterations;
                     }
-                    gradient.addColorStop(stop, fill[j]);
+
+                    // Prevent color stop generation going backwards from floating point imprecision
+                    const clampedStop = Math.max(lastIterationStop, stop);
```
Comment (themoonrat) [inline packages/text/src/Text.ts]: do we also want to make sure that we don't have a point that's greater than 1? Or do you feel that's not possible to achieve even with the rounding issues?

**Comment by TheWyo** [comment]
> Hello @TheWyo ! Thanks for the PR and for the playground
> When I try to use your playground, in the console I see
> 
> "Uncaught TypeError: Failed to execute 'createLinearGradient' on 'CanvasRenderingContext2D': The provided double value is non-finite."
> 
> Could you fix this please? Code looks good, would just like to see for myself :)

My bad, I was curious if the same code worked in v4, didn't realise saving that setting saved it to the same url as opposed to making a copy. Should work for you now.

File: packages/text/src/Text.ts (hunk 1 above)
Comment (TheWyo) [reply packages/text/src/Text.ts]: Trying to add a stop >1 does throw a DOMException from addColorStop apparently, so might be a good idea for safety's sake.

**Pull request merged**